osf_tests/test_comment.py

```diff
@@ -47,173 +53,290 @@ def auth(user):
+    project = ProjectFactory(creator=user)
+    project.add_contributor(contributor)
+    return project
+
+
+@pytest.fixture()
+def comment_self_mentioned(user):
+    return 'This is a comment with a good mention [@Mentioned User](http://localhost:5000/{}/).'.format(user._id)
+
+
+@pytest.fixture()
+def comment_contributor_mentioned(contributor):
+    return 'This is a comment with a good mention [@Mentioned User](http://localhost:5000/{}/).'.format(contributor._id)
+
+
+@pytest.fixture()
+def comment_invalid_user_mentioned():
+    return 'This is a comment with a good mention [@Mentioned User](http://localhost:5000/qwerty/).'
+
+
+@pytest.fixture()
+def comment_too_long():
+    return ''.join(['c' for _ in range(settings.COMMENT_MAXLENGTH + 3)])
+
+
+@pytest.fixture()
+def comment_too_long_with_mention(user):
+    mention = '[@George Ant](http://localhost:5000/{}/)'.format(user._id)
+    return ''.join(['c' for _ in range(settings.COMMENT_MAXLENGTH - 8)]) + mention
+
+
+@pytest.fixture()
+def comment_valid():
+    return 'This is a good comment'
+
+
+@pytest.fixture()
+def comment_mention_valid(contributor):
+    return 'This is a comment [@User](http://localhost:5000/{}/).'.format(contributor._id)
+
+
+@pytest.fixture()
+def comment_mention_project_with_contributor(contributor, project_with_contributor):
+    return 'This is a comment [@User](http://localhost:5000/{}/).'.format(contributor._id)
+
+
+@pytest.fixture()
+def comment_mention_unreg_contributor(unreg_contributor):
+    return 'This is a comment [@Unconfirmed User](http://localhost:5000/{}/).'.format(unreg_contributor._id)
+
+
+@pytest.fixture()
+def comment_mention_non_contributor(user_without_nodes):
+    return 'This is a comment [@User](http://localhost:5000/{}/).'.format(user_without_nodes._id)
+
+
+@pytest.fixture()
+def comment_mention_edited_twice(comment):
+    comment.ever_mentioned.add(comment.user)
+    comment.save()
+    return 'This is a new comment [@User](http://localhost:5000/{}/).'.format(comment.user)
+
+
+@pytest.fixture()
+def comment_mentioned_with_contributors(user):
+    return 'This is a new comment [@User](http://localhost:5000/{}/).'.format(user._id)
+
+
 def test_comments_have_longer_guid():
     comment = CommentFactory()
     assert len(comment._id) == 12
 
+
 def test_comments_are_queryable_by_root_target():
     root_target = ProjectFactory()
     comment = CommentFactory(node=root_target)
     assert Comment.objects.filter(root_target=root_target.guids.first()).first() == comment
 
 
+def pytest_generate_tests(metafunc):
+    # called once per each test function
+    if not metafunc.cls:
+        return
+    if not hasattr(metafunc.cls, 'params'):
+        return
+    funcarglist = metafunc.cls.params.get(metafunc.function.__name__)
+    if not funcarglist:
+        return
+    argnames = sorted(funcarglist[0])
+    metafunc.parametrize(argnames, [[funcargs[name] for name in argnames] for funcargs in funcarglist])
```

Comment:
Neat @Johnetordoff 💯  I hadn't used this before, but it looks like a really slick way to organize the tests - easy to follow, more DRY.. nice work.